### ISSUE TITLE
Statut de l'observation absent dans l'appli et laisse le champ vide dans la synthese

### ISSUE THREAD

**Issue by Amegilla** [PnX-SI/gn_mobile_occtax#66]
Le champs id_nomenclature_observation_status (Présent/Non observé) est absent de l'appli et laissé à null dans la synthese.

**Comment by camillemonchicourt** [comment]
C'est volontaire que le champs soit absent de la saisie.
Quand on fait une occurrence, c'est forcément parce qu'on a observé un taxon. Et pour le mobile on voulait alléger au maximum la saisie.

Même si depuis la version 1.1.0 on peut masquer ou non des champs.

Pour la valeur dans Occtax et dans la Synthèse, ce sont les valeurs par défaut définies dans la BDD qui devraient s'occuper de les renseigner.

De notre côté on a mis "Présent" par défaut, de mémoire.

**Comment by camillemonchicourt** [comment]
OK à vérifier au niveau de GeoNature.
Car pas sur que si le champs n'est pas rempli par Occtax-mobile, alors il soit rempli par les valeurs par défaut définies dans GeoNature.
A priori non comme tu l'indiques.

**Comment by xavyeah39** [comment]
Je confirme ce comportement et le fait que les valeurs par défaut définies dans GeoNature ne sont pas renseignées lors d'une synchro depuis le mobile vers occtax et la synthèse.
Pour les données issues de synchro mobile, on se retrouve donc avec des valeurs NULL dans ces champs (id_nomenclature : observation_status, geo_object_nature, et blurring) au lieu des valeurs par défaut configurées côté GeoNature .

L'effet collatéral, c'est que ces observations ne remontent pas non-plus dans GeoNature-Atlas où par défaut, la vue syntheseff filtre sur les données dont le statut d'observation est "Présent". Contournable en adaptant la vue pour prendre en compte aussi les valeurs NULL mais pas idéal car on a quand même des observations en synthèse avec un remplissage partiel de ces nomenclatures SINP...

L'idéal serait peut-être que les valeurs par défaut définies au niveau de GeoNature puissent se substituer aux valeurs NULL renvoyées par l'appli mobile lorsque certains champs n'y sont pas implémentés ?

**Comment by xavyeah39** [comment]
Je confirme ce comportement et le fait que les valeurs par défaut définies dans GeoNature ne sont pas renseignées lors d'une synchro depuis le mobile vers occtax et la synthèse.
Pour les données issues de synchro mobile, on se retrouve donc avec des valeurs NULL dans ces champs (id_nomenclature : observation_status, geo_object_nature, et blurring) au lieu des valeurs par défaut configurées côté GeoNature .

L'effet collatéral, c'est que ces observations ne remontent pas non-plus dans GeoNature-Atlas où par défaut, la vue syntheseff filtre sur les données dont le statut d'observation est "Présent". Contournable en adaptant la vue pour prendre en compte aussi les valeurs NULL mais pas idéal car on a quand même des observations en synthèse avec un remplissage partiel de ces nomenclatures SINP...

L'idéal serait peut-être que les valeurs par défaut définies au niveau de GeoNature puissent se substituer aux valeurs NULL renvoyées par l'appli mobile lorsque certains champs n'y sont pas implémentés ?

**Comment by camillemonchicourt** [comment]
Les valeurs par défaut dans la synthèse basées sur la table des valeurs par défaut devrait se charger de renseigner ces champs automatiquement quand ils ne sont pas remplis.

**Comment by samuelpriou** [comment]
Bonjour à tous,

Est il prévu de forcer Occtax mobile à donner la valeur 'présent' dès que les données sont envoyée dans GeoNature ? 
La suggestion de @xavyeah39 est intéressante "L'idéal serait peut-être que les valeurs par défaut définies au niveau de GeoNature puissent se substituer aux valeurs NULL renvoyées par l'appli mobile lorsque certains champs n'y sont pas implémentés ?"

Merci

**Comment by PNPyrenees** [comment]
Bonjour,

Je viens d'être confronté à cette anomalie suite à un retour d'intégration des données dans le SINP.
J'en comprends que ce sera corrigé à la prochaine release (sauf si c'est déjà le cas).

Par contre, je me permets de soulever ici la question de pouvoir renseigner des observations négatives via le mobile.
Ça répond à des besoins assez rares mais potentiels. 
ex : dans le cadre d'un de nos protocoles visant à actualiser la connaissance sur la répartition d'une espèce, on fait des recherches de présence de l'espèce dans des cours d'eau. Le fait de renseigner que la prospection sur le secteur a été réalisé mais que l'espèce n'a pas été observée est une donnée presque aussi importante que l'information de présence. Je trouve dommage de ne pas pouvoir utiliser le mobile pour saisir cette info d'autant que la donnée de présence passe par le mobile.

Si on ajoute ce champ dans la version mobile avec une valeur par défaut à "Présent" je ne pense pas que ça alourdisse la saisie, un peu l'interface mais est-ce grave si ça répond à un besoin ?
Ce champ pourrait être intégré dans les "informations complémentaires" (comme en version web).

**Comment by camillemonchicourt** [comment]
Oui si ce champs est masqué par défaut, mais affichable.
Les "occurrences d'absence" restent selon moi un concept tordu, et devraient plutôt être gérées dans des protocoles dédiés, soit en Monitoring, soit avec à minima des notions de zones de prospection, etc...
Mais pas de soucis pour le rendre affichable à ceux qui le souhaiteraient.

**Comment by PNPyrenees** [comment]
J'y pensais en écrivant mon précédent message et je suis d'accord que c'est une notion plus adapté pour monitoring mais c'est aussi plus lourd à mettre en place et dans notre contexte la simple saisie de données d'absence répond au besoin. Et c'est juste pour un rafraichissement ponctuel de la connaissance. Bref... Ça se discute...

Si on peut l'afficher c'est parfait mais je ne vois pas comment faire dans la doc ... 
https://github.com/PnX-SI/gn_mobile_occtax#nomenclature-settings
Il n'apparait pas dans la liste. 
Est-ce juste un oubli ?
Si je met ce qui suit, ça fonctionne ?
[...]
{
    "key": "STATUT_OBS",
    "visible": true,
    "default": true
},
[...]

**Comment by DonovanMaillard** [comment]
Bonjour, 

> Si je met ce qui suit, ça fonctionne ?
[...]
{
"key": "STATUT_OBS",
"visible": true,
"default": true
},
[...]

à tester... de mémoire, non. Contrairement à ce qu'indique Camille, je crois que ce champs là ne peut pas être ajouté. On avait acté de ne pas l'implémenter car le mobile est dédié au relevé d'observations et pas lié à des protocoles avec une possible non-observation. 

En effet, indiquer une non observation nécessite à minima de définir un temps de recherche, une méthode, et une station. Et c'est une notion (bien que présente dans le sinp) qui me pose toujours question... à chaque papillon que j'observe, pourquoi ne pas indiquer les 5300 autres taxons comme étant non observés ?

**Comment by camillemonchicourt** [comment]
Oui on n'avait pas du tout prévu ce champs dans Occtax-mobile qu'on avait voulu simplifier au maximum, mais au final on ajoute petit à petit tous les champs, donc autant ajouter celui-ci aussi, masqué par défaut.

**Comment by DonovanMaillard** [comment]
Oui, dans tous les cas on est en train de faire au maximum que le mobile et le web permettent les mêmes champs, mêmes fonctionnalités et mêmes paramètrages pour garder une cohérence entre les terminaux.

**Comment by Splendens** [comment]
Bonjour,

Des associations m'ont fait remonté le besoin de pouvoir saisir des données d'absence via le mobile également. C'est vrai que serait plutôt du ressort de monitoring, mais le champ du statut d'observation (Présent/Non observé) réponds très bien à plusieurs cas de figure de suivi de populations, sans avoir besoin d'en faire un protocole de monitoring (et en plus, cela permettrait de saisir directement sur le terrain !).

Bref, on me demande si l'ajout de ce champ est prévu à terme (et à quelle échéance) ?

Merci d'avance !

**Comment by camillemonchicourt** [comment]
Rien de financé à ma connaissance actuellement.

Mais n'importe qui peut financer ou réaliser une telle évolution si il le souhaite.

**Comment by AudreyEnGuyane** [comment]
Bonjour à tous, 
Je plussoie le besoin d'avoir la possibilité de savoir des données d'absence. Je sors de séminaires où il était notamment question de PNA ou d'EEE.
Dans ces cadres, il est vraiment nécessaire de pouvoir saisir des données d'absence car l'absence de données n'a pas la même signification qu'une donnée d'absence et cette distinction peut-être cruciale quand on suit/recherche des espèces cibles.

**Comment by sgrimault** [comment]
Bonjour @camillemonchicourt,

Pour l'ajout d'un nouveau champ de la nomenclature `STATUT_OBS`, non visible, non éditable, prenant comme valeur par défaut, celle remontée par l'API, ce dernier est mappé avec l'attribut `id_nomenclature_observation_status` et est situé au même niveau que les autres dans le relevé, comme par exemple `TYP_GRP` mappé dans le relevé avec l'attribut `id_nomenclature_grp_typ` ?

**Comment by camillemonchicourt** [comment]
Non, le statut d'observation est au niveau du taxon, avec l'état biologique, la méthode de détermination, le statut biologique, le comportement...
Voir en web : 

![Image](https://github.com/user-attachments/assets/e30fc8d2-a0c1-43f4-9ef2-91464bdd5627)

Et l'objectif de ce ticket est justement que ce champs (nomenclature) soit désormais saisissable pour ceux qui veulent l'activer.
Mais de ne pas l'activer de base dans la configuration. Que cela soit une action volontaire de l'administrateur que de le proposer à la saisie.

**Comment by camillemonchicourt** [comment]
Le champs est désormais activable à la saisie dans la 2.8.0 (désactivé par défaut).
